dask/dataframe/io/hdf.py

```diff
@@ -385,7 +389,10 @@ def read_hdf(pattern, key, start=0, stop=None, columns=None,
         lock = get_scheduler_lock()
 
     key = key if key.startswith('/') else '/' + key
-    paths = sorted(glob(pattern))
+    if isinstance(pattern, str):
+        paths = sorted(glob(pattern))
+    else:
+        paths = pattern
```

Comment:
This is a totally reasonable way to do it, but note that the functions in dask.bytes effectively do exactly the same, by calling glob on a local filesystem instance - this is so that various storage backends can be used, which is not applicable to reading hdf5.